base_classes/NXbeam.nxdl.xml

```diff
@@ -26,101 +26,220 @@
     xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
     name="NXbeam"
     type="group" extends="NXobject">
-
+    <symbols>
+        <doc>
+            These symbols coordinate datasets with the same shape.
+        </doc>
+        <symbol name="nP">
+            <doc>Number of scan points.</doc>
+        </symbol>
+        <symbol name="m">
+            <doc>Number of channels in the incident beam spectrum, if known</doc>
+        </symbol>
+        <symbol name="c">
+            <doc>Number of moments representing beam divergence (x, y, xy, etc.)</doc>
+        </symbol>
+    </symbols>
     <doc>
         Properties of the neutron or X-ray beam at a given location. 
-        
-        It will be referenced
-        by beamline component groups within the :ref:`NXinstrument` group or by the :ref:`NXsample` group. Note
-        that variables such as the incident energy could be scalar values or arrays. This group is
+
+        This group is referenced
+        by beamline component groups within the :ref:`NXinstrument` group or by the :ref:`NXsample` group. This group is
         especially valuable in storing the results of instrument simulations in which it is useful
         to specify the beam profile, time distribution etc. at each beamline component. Otherwise,
         its most likely use is in the :ref:`NXsample` group in which it defines the results of the neutron
-        scattering by the sample, e.g., energy transfer, polarizations. </doc>
+        scattering by the sample, e.g., energy transfer, polarizations.
+
+        Note that incident_wavelength and related fields can be a scalar values or arrays, depending on the use case.
+        To support these use cases, the explicit dimensionality of these fields is not specified, but it can be inferred
+        by the presense of and shape of accompanying fields, such as incident_wavelength_weights for a polychromatic beam.
+        </doc>
     <field name="distance" type="NX_FLOAT" units="NX_LENGTH">
         <doc>Distance from sample</doc>
     </field>
     <field name="incident_energy" type="NX_FLOAT" units="NX_ENERGY">
         <doc>Energy on entering beamline component</doc>
```

Comment:
I assume this refers to the energy of each particle/photon in the beam, rather than the power of the beam. Might be helpful to make the doc string more specific. Suggest: "Energy carried by each particle of the beam on entering the beamline component"